OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListFragment.kt

```diff
@@ -193,6 +234,7 @@ class MyCoursesListFragment : OfflineSupportBaseFragment(), RefreshListener {
                             fromCache = false
                         )
                     }
+                    finalizeInAppPurchase()
```

Comment:
IMO, `resetPurchase()` is more appropriate as the InAppPurchase flow was already completed at this point, we just need to reset the data and dismiss the loader.